cadc-storage-permissions/src/intTest/java/org/opencadc/inventory/permissions/PermissionsClientTest.java

```diff
@@ -0,0 +1,211 @@
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.inventory.permissions;
+
+import ca.nrc.cadc.auth.SSLUtil;
+import ca.nrc.cadc.util.FileUtil;
+import ca.nrc.cadc.util.Log4jInit;
+import java.net.URI;
+import java.security.PrivilegedExceptionAction;
+import java.util.List;
+import javax.security.auth.Subject;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.opencadc.gms.GroupURI;
+
+public class PermissionsClientTest {
+
+    private static final Logger log = Logger.getLogger(PermissionsClientTest.class);
+
+    // Subject NOT authorized in Baldur.properties to retrieve grant information.
+    private static Subject cadcRegTest1Subject;
+
+    // Subject authorized in Baldur.properties to retrieve grant information.
+    private static Subject cadcAuthTest1Subject;
+
+    private static URI serviceID;
+    private static URI testArtifact;
+    private static GroupURI readGroup1;
+    private static GroupURI writeGroup1;
+    private static GroupURI writeGroup2;
+    private static GroupURI writeGroup3;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        Log4jInit.setLevel("org.opencadc.inventory.permissions", Level.DEBUG);
+        Log4jInit.setLevel("ca.nrc.cadc.", Level.DEBUG);
+
+        cadcRegTest1Subject = SSLUtil.createSubject(
+            FileUtil.getFileFromResource("x509_CADCRegtest1.pem", PermissionsClientTest.class));
+        cadcAuthTest1Subject = SSLUtil.createSubject(
+            FileUtil.getFileFromResource("x509_CADCAuthtest1.pem", PermissionsClientTest.class));
+
+        serviceID = URI.create("ivo://cadc.nrc.ca/baldur");
+
+        testArtifact = URI.create("cadc:TEST-GROUPS/foo");
+        readGroup1 = new GroupURI("ivo://cadc.nrc.ca/gms?TestReadGroup-1");
+        writeGroup1 = new GroupURI("ivo://cadc.nrc.ca/gms?TestWriteGroup-1");
+        writeGroup2 = new GroupURI("ivo://cadc.nrc.ca/gms?TestWriteGroup-2");
+        writeGroup3 = new GroupURI("ivo://cadc.nrc.ca/gms?TestWriteGroup-3");
+    }
+
+    @Test
+    public void testAnonAccess() {
+        try {
+            PermissionsClient testSubject = new PermissionsClient(serviceID);
+            try {
```

Comment:
bad and unnecessary to have a try/catch inside here

Reply:
Fixed.